README.md
Olá, meu nome é **Ariel Gamal** e esta é minha tarefa para a **JusBrasil =D**  
<br>

## Observação:  

Este é um projeto que é necessário iniciar um servidor para que a aplicação funcione,
será necessário dar o comando **npm install** e **npm start** duas vezes. (siga o passo a passo)
<br>
<br>

# Passo a passo para rodar a aplicação.  

## **Passo 1:**
Abra o terminal.  
Crie uma pasta com um nome de sua preferencia.  
Entre na pasta criada com o comando **cd nome-da-pasta**.    
Clone o repositório do github em sua pasta com comando **git clone**.  

*Toda a aplicação será "baixada" em sua maquina.*
<br>
<br>

## **Passo 2:**
Entre na pasta clonada com o comando **cd ArielJusDesafio**.  
A pasta ArielJus possui 2 pastas: 
<ul>
  <li>frontJusTS</li>
  <li>backJus</li>
</ul>
<br>

## **Passo 3:**
Entre na pasta backJus com o comando **cd backJus** e de o comando **npm install**.  
*Após dar o npm install todos o pacotes serão baixados e estará pronto para iniciar o servidor.*
<br>
<br>

## **Passo 4: Iniciando o servidor**
Na mesma pasta backJus use o comando **npm start**  
O servidor será iniciado. 👨‍💻👨‍💻👨‍💻
<br>
<br>

## **Passo 5:**
Deixe esse terminal aberto com o servidor funcionando e abra um outro terminal.
Entre na pasta frontJusTS com o comando **cd frontJusTS**

*OBS: Se voce estiver na pasta backJus é necessário retornar para a pasta ArielJus.*  
*Para isso use o comando **cd..** e depois use o comando **cd frontJusTS***  
De o comando **npm install**

*Após dar o npm install todos o pacotes serão baixados*
<br>
<br>

## **Passo 6: Iniciando a aplicação.**
Na mesma pasta frontJusTs use o comando **npm start**.
A aplicação está pronta para ser utilizada e o navegador irá abrir a página de busca. 👨‍💻👨‍💻👨‍💻
<br>
<br>


# Consultas

**Pode ser realizada a consulta por número de processo:  
Tente os processos:**  
5001682-88.2020.8.13.0672

6021481-77.2010.5.23.0332

7023481-77.2013.5.23.0563

8231431-78.2011.6.11.0772

9011481-41.2000.1.56.0222
<br>
<br>

**Pode ser realizada a consulta por sigla do tribunal:  
Tente as siglas:**  
TJRJ  
TJSP  
TJMG  

**Tente também:**  
Não digitar nada e apertar o botão.

<br>

# RODANDO OS TESTES  
Na pasta **frontJusTS** digite no terminal **npm test**.

HISTORY.md
1 - Resolvi fazer a parte do back end com Node.js por já estar mais familiarizada com a linguagem.
2 - Fiz o front com React usando vite.

2 - De inicio comecei fazendo o front-end, coloquei um header e um footer, mas foi retirado. Achei a aplicação
tao bonita sem nada que preferi retirar.

3- Iniciei o front-end, fiz todos os códigos integrei com o back end, mas estava achando os códigos
muito desorganizados.

 - Uma das coisas que me incomodava foi que estava usando o css com styled components, e me dei conta que achei um pouco feio.

 - Os nomes que eu criei para as tags estavam horriveis e estava me incomodando, quase usei css puro rs. 
 - Masa chei super interessante o tailwind, então melhorei o visual com o tailwind mesmo.

 -Mas ainda sim achei os códigos bem desorganizado:
  Então reiniciei o projeto inteiro com React + vite e Typescript, porém meu conhecimento com type é mais raso, tentei deixar com menos informação possível e tive muita dificuldade de implementar o typescript por não estar muito habituada em usar.

4- Depois de tudo terminado, estou num dilema se coloco novamente o header e o footer, mas vai ficar sem mesmo.

5- A parte do back iniciei com apenas a busca pelo numero do processo, pois achei que seria muito dificil colocar mais de
dois tipos de busca.
Mas depois eu dei um jeito de colocar a busca pelo tribunal.

6- Fiz os testes unitários de renderização com o Vitest.
Como eu usei o Vite para fazer a aplicação, vi que davam alguns erros com o Jest e o react library test. E que tudo se resolveria usando o Vitest junto com o Jest e o react library test.
E no final acabou saindo a mesma coisa porque os códigos são os mesmos.



backJus/Processos/index.js
const processos = [{
  id: 1,
  cnj: "5001682-88.2020.8.13.0672",
  nomeDasPartes: [
    {
      nome: "Paulo Victor",
      posicao: "Advogado envolvido"
    },
    {
      nome: "Ariel Gamal",
      posicao: "Réu"
    },
    {
      nome: "Chuck Norris",
      posicao: "O Juiz"
    },
  ],
  tribunalOrigem: "TJRJ",
  dataInicio:"10-01-2000",
  movimentacoes: [
    {
      data: "10-01-2001",
      descricao: "Este é o processo de cnj: 5001682-88.2020.8.13.0672, é o PRIMEIRO PROCESSO ele é do TRIBUNAL DO RIO DE JANEIRO."
    },
    {
      data: "10-01-2002",
      descricao: "Repare que se voce procurar pela sigla do tribunal do Rio de Janeiro TJRJ teremos 2 resultados de pesquisa."
    },
    {
      data: "10-01-2003",
      descricao: "Suspensão do Prazo Prazo referente ao usuário foi alterado para 20/01/2021 devido à alteração da tabela de feriados."
    },
    {
      data: "10-01-2004",
      descricao: "Chuck Norris não lê livros. Ele os encara até obter a informação que deseja."
    },
    {
      data: "25-06-2005",
      descricao: "Remetido ao DJE Relação: 01/07/2051 Teor do ato: Vistos. Na forma do artigo 513 2º,\
      intime-se a executada, por seu advogado, para que, no prazo de 15(quinze) dias, pague o valor \
      indicado no demonstrativo discriminado e atualizado do crédito, acrescido de custas, se houver. \
      Fica a parte executada advertida de que, transcorrido o prazo previsto no art. 523 sem o pagamento\
      voluntário, inicia-se o prazo de 15(quinze) dias para que, independentemente de penhora ou nova intimação,\
      apresente, nos próprios autos, sua impugnação. Não ocorrendo pagamento voluntário do prazo do artido 523 do CPC,\
      o débito será acrescido de multa de dez por cento e, também, de honorários de advogados de dez por cento.\
      Ademais, não efetuado o pagamento voluntário do prazo de 15(quinze) dias, independentemente de nova intimação\
      do credor, poderá a parte exequente efetuar pedido de pesquisas junto aos sistemas informatizados à disposição\
      do juizo, devendo comprovar o prévio recolhimento das taxas previstas no art.2º, inc.XI, da Lei Estadual 14.838/12,\
      calculadas por cada diligencia a ser efetuada. Por fim, certificado o trânsito em julgado da decisão e transcorrido\
      o prazo do art. 523, mediante o recolhimento das respectivas taxas, a parte exequente poderá requerer diretamente à\
      serventia a expedição de certidão, nos temos do art.517 do CPC, que servirá também aos fins previstos no art.782\
      todos os Códigos de Processo."
    },
  ]
},
{
  id: 2,
  cnj: "6021481-77.2010.5.23.0332",
  nomeDasPartes: [
    {
      nome: "Fedora da Silva",
      posicao: "Exequente"
    },
    {
      nome: "Pedro Paulo",
      posicao: "Advogado envolvido"
    },
    {
      nome: "Marcos Vinicius",
      posicao: "Réu"
    },
  ],
  tribunalOrigem: "TJSP",
  dataInicio:"10-01-2006",
  movimentacoes: [
    {
      data: "09-02-2011",
      descricao: "Este é o processo de cnj: 3021481-77.2010.5.23.0332, é o SEGUNDO PROCESSO ele é do TRIBUNAL DE SÃO PAULO."
    },
    {
      data: "09-02-2012",
      descricao: "Repare que se voce procurar pela sigla do tribunal de SÃO PAULO - TJSP, teremos 2 resultados de pesquisa."
    },
    {
      data: "09-02-2013",
      descricao: "Lorem ipsum dolor sit amet, consectetur adipisci elit, sed eiusmod tempor incidunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur. Quis aute iure reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint obcaecat cupiditat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum."
    },
    {
      data: "09-02-2014",
      descricao: "Suspensão do Prazo Prazo referente ao usuário foi alterado para 20/08/2020 devido à alteração da tabela de feriados."
    },
    {
      data: "09-02-2015",
      descricao: "Digital e Seguro: A gente faz todo atendimento, todo contato por telefone mesmo. Quando o cliente faz realmente questão que seja marcada reunião, a gente marca. Mas 99% dos casos a gente não precisa marcar reunião. Eu acordo e a primeira coisa que eu faço é atualizar o Jusbrasil para ver os casos que tiveram durante a madrugada”"
    }
  ]
},
{
  id: 3,
  cnj: "7023481-77.2013.5.23.0563",
  nomeDasPartes: [
    {
      nome: "Haroldo Gamal",
      posicao: "Advogado de acusação"
    },
    {
      nome: "Tomas De Gusmão",
      posicao: "Advogado envolvido"
    },
    {
      nome: "Wanda de Moraes",
      posicao: "Exequente"
    },
  ],
  tribunalOrigem: "TJRJ",
  dataInicio:"11-11-2011",
  movimentacoes: [
    {
      data: "11-06-2012",
      descricao: "Este é o processo de cnj: 3023481-77.2013.5.23.0563, é o TERCEIRO PROCESSO ele é do TIRBUNAL DO RIO DE JANEIRO"
    },
    {
      data: "12-07-2013",
      descricao: "Repare que se voce procurar pela sigla do tribunal do Rio de Janeiro - TJRJ, teremos 2 resultados de pesquisa."
    },
    {
      data: "12-07-2013",
      descricao: "O tempo não espera por ninguém. A menos que esse homem seja Chuck Norris."
    },
    {
      data: "25-04-2014",
      descricao: "Certidão de Publicação Expedida Relação: 0109/2018 Data de disponibilização: 25-02-21 Data de Publicação: 02-03-21 Número no Diário: 2569 Página:2488/2503"
    },
    {
      data: "12-07-2015",
      descricao: "DE LIBERAÇÃO DE VALORES Nome, vem a juízo requerer alvará judicial com o intuito de realizar o levantamento da quantia existente na conta bancária que tem como titular seu falecido irmão, Nome, onde quando...O Requerente solicita que seja expedido o Alvará para fins de liberação dos valores referentes ao depósito do benefício que recebia o seu falecido irmão, que era aposentado por invalidez....DOS PEDIDOS E DOS REQUERIMENTOS Em suma, requer: a) Procedência da ação com a expedição do Alvará Judicial, autorizando a liberação dos valores depositados, referente ao benefício que era recebido por"
    },
  ]
},
{
  id: 4,
  cnj: "8231431-78.2011.6.11.0772",
  nomeDasPartes: [
    {
      nome: "Fedora da Silva",
      posicao: "Exequente"
    },
    {
      nome: "Eveline Santos",
      posicao: "Advogado envolvido"
    },
    {
      nome: "Juliana Mendes",
      posicao: "Réu"
    },
  ],
  tribunalOrigem: "TJSP",
  dataInicio:"10-01-2020",
  movimentacoes: [
    {
      data: "11-06-2021",
      descricao: "Este é o processo de cnj: 8231431-78.2011.6.11.0772, é o QUARTO PROCESSO ele é do TIRBUNAL DE SÃO PAULO."
    },
    {
      data: "12-07-2022",
      descricao: "Repare que se voce procurar pela sigla do tribunal de SÃO PAULO - TJSP, teremos 2 resultados de pesquisa."
    },
    {
      data: "05-03-2024",
      descricao: "As lágrimas de Chuck Norris curam o câncer. Que pena que ele nunca tenha chorado."
    },
    {
      data: "12-07-2024",
      descricao: "Suspensão do Prazo Prazo referente ao usuário foi alterado para 20/08/2020 devido à alteração da tabela de feriados."
    },
    {
      data: "25-05-2025",
      descricao: "Processo terminado."
    }
  ]
},
{
  id: 5,
  cnj: "9011481-41.2000.1.56.0222",
  nomeDasPartes: [
    {
      nome: "João Pedro",
      posicao: "Apelante"
    },
    {
      nome: "Pedro Paulo",
      posicao: "Advogado envolvido"
    },
    {
      nome: "Maria da Penha",
      posicao: "Advogado envolvido"
    },
    {
      nome: "Marcos Vinicius",
      posicao: "Exequente"
    },
  ],
  tribunalOrigem: "TJMG",
  dataInicio:"10-01-2010",
  movimentacoes: [
    {
      data: "11-06-2017",
      descricao: "Este é o processo de cnj: 8231431-78.2011.6.11.0772, é o QUINTO E ULTIMO PROCESSO ele é do TIRBUNAL DE MINAS GERAIS"
    },
    {
      data: "12-07-2018",
      descricao: "Repare que se voce procurar pela sigla do tribunal de MINAS GERAIS - TJMG, teremos apenas 1 resultado de pesquisa."
    },
    {
      data: "25-04-2019",
      descricao: "Este é o ultimo resultado."
    },
    {
      data: "25-05-2020",
      descricao: "Chuck Norris não dorme. Ele espera.."
    },

    {
      data: "12-07-2021",
      descricao: "Chuck Norris contou até o infinito ... duas vezes."
    },
  ]
},
];

module.exports = {
  processos
}